app/cells/decidim/translation_addons/report_translation_button_cell.rb

```diff
@@ -82,11 +82,16 @@ def field_label(field)
       end
 
       def already_reported?(field)
-        Decidim::TranslationAddons::Report.exists?(decidim_resource_id: model&.id, decidim_user_id: current_user&.id, field_name: field, locale: current_user&.locale)
+        report = Decidim::TranslationAddons::Report.where(decidim_resource_id: model&.id, field_name: field, locale: current_user&.locale).first
+        return false if report.blank?
+
+        detail = Decidim::TranslationAddons::ReportDetail.where(decidim_translation_addons_report_id: report.id, decidim_user_id: current_user&.id)
+        detail.present?
       end
 
       def already_reported_resource?
-        already_reported_fields = Decidim::TranslationAddons::Report.where(decidim_resource_id: model&.id, decidim_user_id: current_user&.id, locale: current_user&.locale).count
+        already_reported_fields = Decidim::TranslationAddons::Report.joins(:details).where(decidim_resource_id: model&.id,
```

Comment:
polymorphism !! you have a false positive. 
```suggestion
        already_reported_fields = Decidim::TranslationAddons::Report.joins(:details).where(decidim_resource_id: model&.id,
```